discord/message.py

```diff
@@ -237,6 +238,49 @@ def cached_message(self):
         """Optional[:class:`Message`]: The cached message, if found in the internal message cache."""
         return self._state._get_message(self.message_id)
 
+    async def fetch_message(self):
+        """|coro|
+
+        Retrieves the :class:`Message` from the destination.
+
+        This can only be used by bot accounts.
+
+        .. versionadded:: 2.0
+
+        Raises
+        --------
+        ~discord.NotFound
+            The message was not found.
+        ~discord.ChannelNotFound
+            The message's channel was not found.
+        ~discord.Forbidden
+            You do not have the permissions required to get a message.
+        ~discord.HTTPException
+            Retrieving the message failed.
+
+        Returns
+        --------
+        :class:`Message`
+            The message asked for.
+        """
+
+        state = self._state
+        if self.guild_id:
+            guild = state.get_guild(self.guild_id)
+            if guild:
+                channel = guild.get_channel(self.channel_id)
+            else:
+                channel = None
+        else:
+            channel = state.get_channel(self.channel_id)
+
+        data = await state.http.get_message(self.channel_id, self.message_id)
+        if channel is None:
+            channel = Object(id=self.channel_id)
+            if self.guild_id:
+                channel.guild = Object(id=self.guild_id)
```

Comment:
Wouldn't this cause issues as it is assumed that `TextChannel.guild` is `Guild` not just an `Object`, so if a user were to access attributes other than `id` they would get an error?

Reply:
It's a valid complaint. In v2.0 this would be swapped out with a different type named `UnavailableGuild` since the guild model would be split into two. Though that idea is not exactly finalised either.